pkg/daemons/agent/agent_windows.go

```diff
@@ -38,86 +38,64 @@ func kubeProxyArgs(cfg *config.Agent) map[string]string {
 	return argsMap
 }
 
-func kubeletArgs(cfg *config.Agent) map[string]string {
-	bindAddress := "127.0.0.1"
-	_, IPv6only, _ := util.GetFirstString([]string{cfg.NodeIP})
-	if IPv6only {
-		bindAddress = "::1"
+// kubeletArgsAndConfig generates default kubelet args and configuration.
+// Kubelet config is frustratingly split across deprecated CLI flags that raise warnings if you use them,
+// and a structured configuration file that upstream does not provide a convienent way to initailize with default values.
+// The defaults and our desired config also vary by OS.
+func kubeletArgsAndConfig(cfg *config.Agent) (map[string]string, *kubeletconfig.KubeletConfiguration, error) {
+	defaultConfig, err := defaultKubeletConfig(cfg)
```

Comment:
Is this going to work on Windows? Won't stuff like the cgroup driver fail/be unnecessary on this OS?

Reply:
yeah I'm honestly not sure what the windows kubelet does if you try to set the cgroup stuff. I'll try this out in RKE2 before merging.